Mathlib/Combinatorics/Optimization/ValuedCSP.lean

```diff
@@ -53,6 +53,7 @@ def ValuedCsp.Term.evalSolution {Γ : ValuedCsp D C} {ι : Type*}
   t.f (x ∘ t.app)
 
 /-- A valued CSP instance over the template `Γ` with variables indexed by `ι`.-/
+@[reducible]
 def ValuedCsp.Instance (Γ : ValuedCsp D C) (ι : Type*) : Type _ :=
```

Comment:
Instead of reducible, you can change `def` to `abbrev`

Reply:
Are there any general recommendations about when to use `abbrev`? For comparison, `ValuedCsp` is `reducible` `def`.

Reply:
I think `@[reducible] def` was a lean3-ism cause by `abbrev`(`iation`) generating bad docs and no equation lemmas. That flaw is absent in Lean 4.

Reply:
Should both `ValuedCsp` and `ValuedCsp.Instance` be `abbrev` then? Will dot notation work exactly the same?

Reply:
Yes, basically nothing will care.